### PR TITLE
[feature/cumulative--10.0.0] AB#33499 - Deprecated Header Component

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

File: docs/src/dataDisplay/divider/divider.jsx [hunk 1]
```diff
@@ -123,6 +123,30 @@ export default class RelaxedSample extends React.Component {
 }`;
 
 export default class ElementsDivider extends React.Component {
+    onBlur() {
+        console.log('Are you sure you are done?');
```
Comment (groberts314) [inline docs/src/dataDisplay/divider/divider.jsx:127]: Should we add ESLINT suppress for `console.log()` and `window.alert()` calls?

File: docs/src/dataDisplay/list/list.jsx [hunk 2]
```diff
@@ -95,24 +94,25 @@ export default class ElementsList extends React.Component {
...
+                    <Heading variant="h2">
                         List
-                        <Header.Subheader>
-                            A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
```
Comment (groberts314) [inline docs/src/dataDisplay/list/list.jsx:108]: All of the instances of "it's" in here should be "its" (no apostrophe).
Why is this talking about a Card on the List documentation page?  Is that on purpose?  Or was this a copy-and-paste error?

File: docs/src/dataDisplay/list/list.jsx [hunk 3]
```diff
@@ -121,28 +121,29 @@ export default class ElementsList extends React.Component {
...
+                    <Heading variant="h2">
                         Horizontal
-                        <Header.Subheader>
-                            A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
```
Comment (groberts314) [inline docs/src/dataDisplay/list/list.jsx:129]: "it's" => "its" here as well.
Actually ... was this just copied and pasted from above...?  Is it even relevant here?  Copy-and-paste error?  Same goes for several additional placements below.

File: docs/src/layout/page/demo/demo.jsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import {
     AppBar,
     Dropdown,
-    Header,
+    Typography,
```
Comment (groberts314) [inline docs/src/layout/page/demo/demo.jsx:4]: Are we not using this?  Only the import had to change, but no actual code? 😕 
If we're not using it, let's just remove it, yeah?

File: docs/src/surfaces/card/card.jsx [hunk 5]
```diff
@@ -275,21 +278,22 @@ export default class ViewsTitleBar extends React.Component {
...
+                    <Heading variant="h2">
                         Card
-                        <Header.Subheader>
-                            A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        A basic card has a faint border wrapping it's contents. It's box model is fluid so that it can adapt to it's parent container's width.
```
Comment (groberts314) [inline docs/src/surfaces/card/card.jsx:292]: Hey, here's the original! 😁 
Let's fix up the "it's" => "its" issue!

File: docs/src/surfaces/card/card.jsx [hunk 6]
```diff
@@ -309,22 +313,23 @@ export default class ViewsTitleBar extends React.Component {
...
-                            <code>color(backgroundColorNest)</code>
-                            .
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        A card can give the appearance of being nested. The parent's background color needs to be set to
+                        {' '}
+                        <code>color(backgroundColorNest)</code>
```
Comment (groberts314) [inline docs/src/surfaces/card/card.jsx:323]: Is this still useful?  Or should we start talking about theme/palette here?

File: docs/src/surfaces/card/card.jsx [hunk 7]
```diff
@@ -343,12 +348,13 @@ export default class ViewsTitleBar extends React.Component {
...
+                    <Heading variant="h2">
                         Compact
-                        <Header.Subheader>
-                            A card can have it's padding reduced.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        A card can have it's padding reduced.
```
Comment (groberts314) [inline docs/src/surfaces/card/card.jsx:356]: "it's" => "its".  And capitalize Card.

File: docs/src/surfaces/titleBar/titleBar.jsx [hunk 8]
```diff
@@ -139,12 +141,13 @@ export default class ViewsTitleBar extends React.Component {
...
+                    <Heading variant="h2">
                         Sub Title
-                        <Header.Subheader>
-                            Title Bar's can have a sub title.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
+
+                    <Typography variant="body1">
+                        Title Bar's can have a sub title.
```
Comment (groberts314) [inline docs/src/surfaces/titleBar/titleBar.jsx:149]: I don't think we need the apostrophe here.
> Title Bars can have a sub title.

**Review by groberts314 — CHANGES_REQUESTED** [review]
A few things, mostly on Docs content, which was unfortunately really visible in this PR 😁 
If necessary, we can negotiate on what to fix now versus later.

File: docs/src/layout/page/demo/demo.jsx (hunk 4 above)
Comment (cbrewer1689) [reply docs/src/layout/page/demo/demo.jsx]: We weren't using Header it looks like, but we are using Typography.

File: docs/src/surfaces/card/card.jsx (hunk 6 above)
Comment (cbrewer1689) [reply docs/src/surfaces/card/card.jsx:323]: Should talk about theme, eventually.

File: docs/src/surfaces/titleBar/titleBar.jsx (hunk 8 above)
Comment (cbrewer1689) [reply docs/src/surfaces/titleBar/titleBar.jsx]: Fixed.

File: docs/src/surfaces/card/card.jsx (hunk 7 above)
Comment (cbrewer1689) [reply docs/src/surfaces/card/card.jsx]: Fixed.

File: docs/src/surfaces/card/card.jsx (hunk 5 above)
Comment (cbrewer1689) [reply docs/src/surfaces/card/card.jsx]: Fixed.

File: docs/src/dataDisplay/list/list.jsx (hunk 2 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/list/list.jsx]: Looks like a copy and paste error.

File: docs/src/dataDisplay/divider/divider.jsx (hunk 1 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/divider/divider.jsx:128]: Fixed.

File: docs/src/dataDisplay/list/list.jsx (hunk 3 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/list/list.jsx]: Fixed.

File: docs/src/dataDisplay/divider/divider.jsx [hunk 9]
```diff
@@ -123,6 +123,35 @@ export default class RelaxedSample extends React.Component {
...
+    }
+
+    onClick() {
+        // eslint-disable-next-line no-alert
+        window.alert('You did it! You clicked the text area.');
+    }
+
+    onFocus() {
+        // eslint-disable-next-line no-console
+        console.log('Your focus determins your reality.');
```
Comment (groberts314) [inline docs/src/dataDisplay/divider/divider.jsx:142]: "determines" is misspelled 😁

File: docs/src/dataDisplay/divider/divider.jsx (hunk 1 above)
Comment (groberts314) [reply docs/src/dataDisplay/divider/divider.jsx:128]: Looks good.  Resolving.

File: docs/src/dataDisplay/list/list.jsx (hunk 2 above)
Comment (groberts314) [reply docs/src/dataDisplay/list/list.jsx]: Seems to have been removed.  Resolving.

File: docs/src/dataDisplay/list/list.jsx (hunk 3 above)
Comment (groberts314) [reply docs/src/dataDisplay/list/list.jsx]: Looks to have been removed.  Resolving.

File: docs/src/inputs/checkbox/checkbox.jsx [hunk 10]
```diff
@@ -105,4 +105,4 @@ function DocsCheckbox(props) {
 
 DocsCheckbox.propTypes = propTypes;
 
-export default DocsCheckbox;
+export default DocsCheckbox;
```
Comment (groberts314) [inline docs/src/inputs/checkbox/checkbox.jsx:108]: Please revert this change and bring back my terminating newline :-)

File: docs/src/inputs/checkbox/devSandbox/devSandbox.jsx [hunk 11]
```diff
@@ -253,26 +278,33 @@ export default class ElementsCheckbox extends React.Component {
...
+                    >
                         Label Click
-                        <Header.Subheader>
-                            Sometimes you may want to disable a checkbox from being checked by clicking it's label definition.
-                        </Header.Subheader>
-                    </Header>
+                    </Heading>
 
+                    <Typography>
+                        Sometimes you may want to disable a checkbox from being checked by clicking it's label definition.
```
Comment (groberts314) [inline docs/src/inputs/checkbox/devSandbox/devSandbox.jsx:289]: Wrong "its" here (should not have an apostrophe) but non-blocking as it was like this and wasn't really changed as part of this PR.

File: docs/src/inputs/segmentedControls/segmentedControls.jsx [hunk 12]
```diff
@@ -171,18 +171,19 @@ export default class CollectionsSegmentedControls extends React.Component {
...
                     <Card>
-                        <Header size="large">Props</Header>
+                        <Typography size="large">Props</Typography>
 
                         <TableProps props={props} />
                     </Card>
 
                     {/* Segmented Controls */}
-                    <Header size="large" style={{ marginTop: '55px' }} sub={true}>
+                    <Typography size="large" style={{ marginTop: '55px' }} sub={true}>
```
Comment (groberts314) [inline docs/src/inputs/segmentedControls/segmentedControls.jsx:180]: I imagine we no longer need `sub={true}` here, right?

File: docs/src/inputs/textArea/textArea.jsx [hunk 13]
```diff
@@ -485,18 +485,19 @@ export default class ElementsTextArea extends React.Component {
...
                     <Card>
-                        <Header size="large">Props</Header>
+                        <Typography size="large">Props</Typography>
 
                         <TableProps props={props} />
                     </Card>
 
                     {/* Text Area */}
-                    <Header size="large" style={{ marginTop: '55px' }} sub >
+                    <Typography size="large" style={{ marginTop: '55px' }} sub >
```
Comment (groberts314) [inline docs/src/inputs/textArea/textArea.jsx:494]: Same ... let's nix the `sub`s if possible.  (Sounds like I'm suggesting anti-submarine warfare ;-)

File: docs/src/layout/page/demo/demo.jsx (hunk 4 above)
Comment (groberts314) [reply docs/src/layout/page/demo/demo.jsx]: The updated diff now shows `Typography` as already there, with some other removals only.  Whereas the diff I originally commented on showed  `Typography` being _added_, without adding corresponding usage in the component's `render()`, which, in my defense, seems suspicious.  I imagine this is likely a product of changing this file in various different 10.0.0 feature branches.

I'll go ahead and resolve this thread.  Nothing to see here ... move along.

File: docs/src/layout/page/page.jsx [hunk 14]
```diff
@@ -147,89 +147,94 @@ function ModulesPage() {
...
+                    <span>
+                        <p style={{ marginTop: 0 }}>
+                            This is a template component, meaning it facilitates in styling a basic page,
+                            excluding the side navigation and TitleBar, by using its sub components.
+                        </p>
+
+                        <p>
+                            It can also wrap other atoms, molecules, and organisms that are non Page sub
+                            components to build up a page. But by using these types of components, Page may not
+                            neccearily know how to apply logic and styling to them.
```
Comment (groberts314) [inline docs/src/layout/page/page.jsx:164]: "necessarily" is misspelled here.

File: docs/src/surfaces/card/card.jsx [hunk 15]
```diff
@@ -42,7 +41,7 @@ export default class NestedCardSample extends React.Component {
     render() {
         return (
             <Card nest={true}>
```
Comment (groberts314) [inline docs/src/surfaces/card/card.jsx:43]: Should just be `nest` instead of `nest={true}`.  Non-Blocking, but I'd have thought ESLINT would be buggin' ;-)

File: docs/src/surfaces/card/card.jsx [hunk 16]
```diff
@@ -72,10 +71,11 @@ export default class CompactSample extends React.Component {
                     </Grid.Column>
 
                     <Grid.Column style={{ padding: '0 0 0 22px', width: 'auto' }}>
-                        <Header size="small" sub={true}>
+                        <Typography size="small" sub={true}>
```
Comment (groberts314) [inline docs/src/surfaces/card/card.jsx:74]: Torpedo those `sub`s!

File: docs/src/surfaces/card/card.jsx (hunk 6 above)
Comment (groberts314) [reply docs/src/surfaces/card/card.jsx:323]: This is non-blocking for now.  Resolving.

File: docs/src/surfaces/card/card.jsx (hunk 5 above)
Comment (groberts314) [reply docs/src/surfaces/card/card.jsx]: Looks to have been fixed as requested (definitely here, and in other places I notice as I'm scanning the updated version of this file).  Resolving.

File: docs/src/surfaces/drawer/drawerActionBar.jsx [hunk 17]
```diff
@@ -1,5 +1,5 @@
 import {
-    Button, Card, Drawer, Header, Icon, Input,
+    Button, Card, Drawer, Typography, Icon, Input,
```
Comment (groberts314) [inline docs/src/surfaces/drawer/drawerActionBar.jsx:2]: Non-blocking but should be on separate lines.

File: docs/src/surfaces/drawer/drawerActionBar.jsx [hunk 18]
```diff
@@ -307,18 +307,19 @@ class ModulesDrawerActionBar extends React.Component {
...
                     <Card>
-                        <Header size="large">Props</Header>
+                        <Typography size="large">Props</Typography>
 
                         <TableProps props={props} />
                     </Card>
 
                     {/* Action Bar */}
-                    <Header anchor="drawer" size="large" style={{ marginTop: '55px' }} sub>
+                    <Typography anchor="drawer" variant="h2" style={{ marginTop: '55px' }} sub>
```
Comment (groberts314) [inline docs/src/surfaces/drawer/drawerActionBar.jsx:316]: Ping ... sonar found another `sub`

File: docs/src/surfaces/card/card.jsx (hunk 7 above)
Comment (groberts314) [reply docs/src/surfaces/card/card.jsx]: "its" is fixed.  "card" is still not capitalized, but meh.  Resolving.

File: docs/src/surfaces/titleBar/titleBar.jsx (hunk 8 above)
Comment (groberts314) [reply docs/src/surfaces/titleBar/titleBar.jsx]: Looks good.  Resolving.

**Review by groberts314 — COMMENTED** [review]
Completed another review pass.  There are a few more things, but I am willing to be flexible on blockers versus NABs so as not to impede forward progress unnecessarily.

**Comment by groberts314** [comment]
Oh nos!  Conflicts! ;-)

File: docs/src/dataDisplay/divider/divider.jsx (hunk 9 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/divider/divider.jsx:142]: 😬

File: docs/src/inputs/segmentedControls/segmentedControls.jsx (hunk 12 above)
Comment (cbrewer1689) [reply docs/src/inputs/segmentedControls/segmentedControls.jsx:180]: MMM, let me see if I can do a find a replace on this.

File: docs/src/surfaces/drawer/drawerActionBar.jsx (hunk 18 above)
Comment (cbrewer1689) [reply docs/src/surfaces/drawer/drawerActionBar.jsx]: Fixed

File: docs/src/surfaces/drawer/drawerActionBar.jsx (hunk 17 above)
Comment (cbrewer1689) [reply docs/src/surfaces/drawer/drawerActionBar.jsx]: Fixed.

File: docs/src/surfaces/card/card.jsx (hunk 16 above)
Comment (cbrewer1689) [reply docs/src/surfaces/card/card.jsx:74]: Fixed.

File: docs/src/surfaces/card/card.jsx (hunk 15 above)
Comment (cbrewer1689) [reply docs/src/surfaces/card/card.jsx]: Fixed.

File: docs/src/layout/page/page.jsx (hunk 14 above)
Comment (cbrewer1689) [reply docs/src/layout/page/page.jsx]: Fixed.

File: docs/src/inputs/textArea/textArea.jsx (hunk 13 above)
Comment (cbrewer1689) [reply docs/src/inputs/textArea/textArea.jsx:494]: Fixed.

File: docs/src/inputs/segmentedControls/segmentedControls.jsx (hunk 12 above)
Comment (cbrewer1689) [reply docs/src/inputs/segmentedControls/segmentedControls.jsx]: Fixed.

File: docs/src/inputs/checkbox/devSandbox/devSandbox.jsx (hunk 11 above)
Comment (cbrewer1689) [reply docs/src/inputs/checkbox/devSandbox/devSandbox.jsx]: Fixed.

File: docs/src/inputs/checkbox/checkbox.jsx (hunk 10 above)
Comment (cbrewer1689) [reply docs/src/inputs/checkbox/checkbox.jsx]: Fixed.

File: docs/src/dataDisplay/divider/divider.jsx (hunk 9 above)
Comment (cbrewer1689) [reply docs/src/dataDisplay/divider/divider.jsx]: Fixed.

Comment (groberts314) [reply docs/src/dataDisplay/divider/divider.jsx]: Looks good.  Resolving.

File: docs/src/inputs/checkbox/checkbox.jsx (hunk 10 above)
Comment (groberts314) [reply docs/src/inputs/checkbox/checkbox.jsx]: Yep, I don't see this file diffed anymore.  Resolving.

File: docs/src/inputs/checkbox/devSandbox/devSandbox.jsx (hunk 11 above)
Comment (groberts314) [reply docs/src/inputs/checkbox/devSandbox/devSandbox.jsx]: Resolving.

File: docs/src/inputs/segmentedControls/segmentedControls.jsx (hunk 12 above)
Comment (groberts314) [reply docs/src/inputs/segmentedControls/segmentedControls.jsx]: Looks good.  Resolving.

File: docs/src/inputs/textArea/textArea.jsx (hunk 13 above)
Comment (groberts314) [reply docs/src/inputs/textArea/textArea.jsx:494]: `sub`s all seem to have been eliminated.  Resolving.

File: docs/src/layout/page/page.jsx (hunk 14 above)
Comment (groberts314) [reply docs/src/layout/page/page.jsx]: Yep, looks good.  Resolving.

File: docs/src/surfaces/card/card.jsx (hunk 15 above)
Comment (groberts314) [reply docs/src/surfaces/card/card.jsx]: Looks good.  Resolving.

File: docs/src/surfaces/card/card.jsx (hunk 16 above)
Comment (groberts314) [reply docs/src/surfaces/card/card.jsx:74]: Resolving.

File: docs/src/surfaces/drawer/drawerActionBar.jsx (hunk 17 above)
Comment (groberts314) [reply docs/src/surfaces/drawer/drawerActionBar.jsx]: Thank you sir!  Resolving.

File: docs/src/surfaces/drawer/drawerActionBar.jsx (hunk 18 above)
Comment (groberts314) [reply docs/src/surfaces/drawer/drawerActionBar.jsx]: Resolving.

**Review by groberts314 — APPROVED** [review]
